lib/Site.js

```diff
@@ -74,6 +77,37 @@ function setExtension(filename, ext) {
   );
 }
 
+/**
+* Creates a function that delays invoking `func` until after `wait` milliseconds have elapsed
+* and the running `func` has resolved/rejected.
+* @param func the promise-returning function to delay
```

Comment:
`func`'s parameter must be an array, right?

Reply:
Nope but we can change it that way
* Currently, it is using `func.apply(context, pendingArgs);` and `pendingArgs` is supposed to be the argument list. So all the delayed functions take in a variable number of arguments `...filePaths`
* Will you prefer it to be `func.apply(context, [pendingArgs]);` instead?

Reply:
[`Function.prototype.apply()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Function/apply) *calls a function with arguments provided as an array*.
The way before https://github.com/MarkBind/markbind-cli/pull/31/commits/55b874aee100d6ecbade3c4927b219ed1eb9451a is preferable, and already the case where `func`'s parameter was an array.
I meant the jsdoc should specify that.

I don't understand why `func.apply(context, [pendingArgs]);` was suggested.

Reply:
Example
```js
function sum(a, b) { // does not take in an array
    return a + b;
}

function wrapsum(a, b) {
     // invalid without [], passed it as sum(a,b) instead of sum([a, b])
    return sum.apply(this, [a, b]); 
}
```

If the delayed functions do not take in a variable number of arguments `...filePaths`, if there is only one path, it will call `rebuildSourceFile("filename")` which won't work as an array is expected

Reply:
That doesn't seem like a suitable example.

In our case, the wrapped function takes in a single argument — either a filename, or an array.

```js
var _pendingArgs = [];
var _sum = 0;

function sum(a) { // does not take in an array
    _sum += a;
    return _sum;
}

function wrapsum(...args) {
    _pendingArgs.push(...args);

    // Fake delay
    if (_pendingArgs < 2) return -1;
    const pendingArgs = _pendingArgs;
    _pendingArgs = [];

    return sum.apply(this, [pendingArgs]); 
}
```

I mean, if the "original" function only expects only 1 number as above, this won't work:

```js
wrapsum(1); // -1
wrapsum(2); // "01,2"
```

If its parameter is an array:

```js
var _sum = 0;

function sum(arr) { // takes in an array
    arr.forEach(a => { _sum += a });
    return _sum;
}
```
Then this will work:

```js
wrapsum(1); // -1
wrapsum(2); // 3
```

Now I understand why `func.apply(context, [pendingArgs]);` was suggested.

Reply:
> Now I understand why `func.apply(context, [pendingArgs]);` was suggested.

I see that it's not necessary if *rest parameters* are used.